src/__swaps__/screens/Swap/components/TokenList/ChainSelection.tsx

```diff
@@ -0,0 +1,248 @@
+  const { accentColor: accountColor } = useAccountAccentColor();
+  const { SwapInputController, userAssetFilter } = useSwapContext();
+  const red = useForegroundColor('red');
+
+  const accentColor = useMemo(() => {
+    if (c.contrast(accountColor, isDarkMode ? '#191A1C' : globalColors.white100) < (isDarkMode ? 2.125 : 1.5)) {
+      const shiftedColor = isDarkMode ? c(accountColor).brighten(1).saturate(0.5).css() : c(accountColor).darken(0.5).saturate(0.5).css();
+      return shiftedColor;
+    } else {
+      return accountColor;
+    }
+  }, [accountColor, isDarkMode]);
+
+  const propToSet = output ? SwapInputController.outputChainId : userAssetFilter;
+
+  const chainName = useSharedValue(
+    propToSet.value === 'all' ? allText : propToSet.value === ChainId.mainnet ? 'ethereum' : chainNameFromChainIdWorklet(propToSet.value)
+  );
+
+  useAnimatedReaction(
+    () => ({
+      outputChainId: SwapInputController.outputChainId.value,
+      userAssetFilter: userAssetFilter.value,
+    }),
+    current => {
+      if (output) {
+        chainName.value = current.outputChainId === ChainId.mainnet ? 'ethereum' : chainNameFromChainIdWorklet(current.outputChainId);
+      } else {
+        chainName.value =
+          current.userAssetFilter === 'all'
+            ? allText
+            : current.userAssetFilter === ChainId.mainnet
+              ? 'ethereum'
+              : chainNameFromChainIdWorklet(current.userAssetFilter);
+      }
+    }
+  );
+
+  const handleSelectChain = useCallback(
+    ({ nativeEvent: { actionKey } }: Omit<OnPressMenuItemEventObject, 'isUsingActionSheetFallback'>) => {
+      runOnUI((set: SharedValue<ChainId | UserAssetFilter>) => {
+        if (actionKey === 'all') {
+          set.value = 'all';
+        } else {
+          set.value = Number(actionKey) as ChainId;
+        }
+      })(propToSet);
+    },
+    [propToSet]
+  );
+
+  const menuConfig = useMemo(() => {
+    const supportedChains = SUPPORTED_CHAINS({ testnetMode: false }).map(chain => {
+      const network = ethereumUtils.getNetworkFromChainId(chain.id);
+
+      const getIconValueForNetwork = (network: Network) => {
+        let name = network;
+        if (network === 'mainnet') {
+          name = 'ethereum' as Network;
+        }
+
+        return `${name}Badge${isDarkMode ? 'Dark' : ''}`;
+      };
+
+      const getActionTitleForChain = (chainId: ChainId) => {
+        if (chainId === ChainId.mainnet) {
+          return 'Ethereum';
+        }
+
+        return chainNameFromChainId(chainId).charAt(0).toUpperCase() + chainNameFromChainId(chainId).slice(1);
+      };
+
+      return {
+        actionKey: `${chain.id}`,
+        actionTitle: getActionTitleForChain(chain.id),
+        icon: {
+          iconType: 'ASSET',
+          iconValue: getIconValueForNetwork(network),
+        },
+      };
+    });
+
+    if (!output) {
+      supportedChains.unshift({
+        actionKey: 'all',
+        actionTitle: 'All Networks' as ChainName,
```

Comment:
mainly just mentioning cause theres a bunch of these but i18n gonna be annoying to do at the end